src/tasks/task-joint-posVelAcc-bounds.cpp

```diff
@@ -176,226 +195,182 @@ namespace tsid
-              }
-              else
-              {
-                  if(verbose == true){
-                    std::cout << "WARNING  qa[i]==m_qMin[i] for joint" << i << std::endl;
-                    std::cout << "You are goting to violate the position bound " << i << std::endl;
-                  }
-                  m_ddqLBPos[i] = 0.0;
-              }
+            m_ddqUBPos[i]  = m_ddqMax_q3[i];
+          }
+          else if(qa[i]!=m_qMax[i])
+          {
+            m_ddqMax_q2[i] = -(dqa[i]*dqa[i])/(2*(m_qMax[i]-qa[i]));
+            m_ddqUBPos[i]  = std::min(m_ddqMax_q2[i],m_minus_dq_over_dt[i]);
           }
           else
           {
-              m_ddqLBPos[i]  = ddqMin_q3[i];
-              if(ddqMax_q3[i] > minus_dq_over_dt[i])
-              {
-                  m_ddqUBPos[i]  = ddqMax_q3[i];
-              }
-              else if(qa[i]!=m_qMax[i])
-              {
-                  ddqMax_q2[i] = -std::pow(dqa[i],2)/(2*(m_qMax[i]-qa[i]));
-                  m_ddqUBPos[i]  = std::min(ddqMax_q2[i],minus_dq_over_dt[i]);
-              }
-              else
-              {
-                  if(verbose == true){
-                    std::cout << "WARNING  qa[i]==m_qMax[i] for joint" << i << std::endl;
-                    std::cout << "You are goting to violate the position bound " << i << std::endl;
-                  }
-                  m_ddqUBPos[i] = 0.0;
-              }
+            if(verbose == true)
+            {
+              std::cout << "WARNING  qa[i]==m_qMax[i] for joint" << i << std::endl;
+              std::cout << "You are goting to violate the position bound " << i << std::endl;
+            }
+            m_ddqUBPos[i] = 0.0;
           }
-       }
+        }
+      }
     }
-    void TaskJointPosVelAccBounds::computeAccLimitsFromViability(const Vector& q,
-                                                                const Vector& dq,
+    void TaskJointPosVelAccBounds::computeAccLimitsFromViability(const Vector& qa,
+                                                                const Vector& dqa,
                                                                 bool verbose)
     {
-        Vector qa = q.tail(m_na);
-        Vector dqa = dq.tail(m_na);
-        double dt_square = std::pow(m_dt,2);
-        Vector dt_dq = m_dt*dqa;
-        Vector minus_dq_over_dt = -dqa/m_dt;
-        Vector dt_two_dq = 2*dt_dq;
-        Vector two_ddqMax = 2*m_ddqMax;
-        Vector dt_ddqMax_dt = m_ddqMax*dt_square;
-        double dq_square = dqa.dot(dqa);
-        Vector q_plus_dt_dq = qa + dt_dq;
-        double two_a = 2*dt_square;
-        Vector b = dt_two_dq + dt_ddqMax_dt;
-        Vector bl = dt_two_dq - dt_ddqMax_dt;
-        for(int i=0; i<m_na; i++)
+      m_dt_dq = m_dt*dqa;
+      m_minus_dq_over_dt = -dqa/m_dt;
+      m_dt_two_dq = 2*m_dt_dq;
+      m_two_ddqMax = 2*m_ddqMax;
+      m_dt_ddqMax_dt = m_ddqMax*m_dt_square;
+      m_dq_square = dqa.dot(dqa);
```

Comment:
just to check, are you sure you shouldn't use cwiseProduct here instead of dot?

Reply:
Yes that is correct it was cwiseProduct